### DISCUSSION TITLE
How do I send the form data to an email recipient?

### DISCUSSION THREAD

**Discussion by SRD75** [react-hook-form/react-hook-form#8209] [Q&A]
Nothing (that I can see, or search for on google) on https://react-hook-form.com/ discusses how to send the form data to an email recipient.

Help appreciated.

**Comment by Moshyfawn** [comment]
Can you elaborate on what you want to achieve?

Any code examples?

**Comment by Moshyfawn** [comment]
You get the parsed to object form `data` as the first arg of you submit handler and the submission `event` object as the second _(where you can get the Form Data object)_. It's up to you to do whatever with the submitted data on submission

**Comment by SRD75** [comment]
I would love code examples of a form handler emailing me the data from the form, so that the form functions as a contact form.

**Reply by Moshyfawn** [reply]
There's nothing extraordinary that's different from any other form. [Here's a CSB](https://codesandbox.io/s/great-lumiere-80rgtr?file=%2Fsrc%2FApp.tsx).

### TLDR
```jsx
const onSubmit = async (data, event) => {
  let formData = new FormData(event?.target)
  // INFO: here goes your "sending email" method implementation
  await promise(1000, formData)
}

const App = () => {
  const { handleSubmit, register, formState } = useForm()

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label htmlFor="email">Email</label>
      <input id="email" {...register('email')} />
      <label htmlFor="title">Title</label>
      <input id="title" {...register('title')} />
      <label htmlFor="body">Body</label>
      <textarea id="body" {...register('body')} />
      <button type="submit">
        Submit
      </button>
    </form>
  )
}

export default App
```

I'm not gonna implement the "send email" API as it's way out of scope of RHF and it's really up to you how it happens.
